microprofile/fault-tolerance/src/test/java/io/helidon/microprofile/faulttolerance/BulkheadBean.java

```diff
@@ -37,30 +39,52 @@ static class ConcurrencyCounter {
         private int concurrentCalls;
         private int totalCalls;
 
-        synchronized void increment() {
-            currentCalls++;
-            if (currentCalls > concurrentCalls) {
-                concurrentCalls = currentCalls;
-            }
-            totalCalls++;
+        private final Semaphore accessGuard = new Semaphore(1, true);
+
+        void increment() {
+            access(() -> {
+                currentCalls++;
+                if (currentCalls > concurrentCalls) {
+                    concurrentCalls = currentCalls;
+                }
+                totalCalls++;
+                return null;
+            });
+        }
+
+        void decrement() {
+            access(() -> {
+                currentCalls--;
+                return null;
+            });
         }
 
-        synchronized void decrement() {
-            currentCalls--;
+        int concurrentCalls() {
+            return access(() -> concurrentCalls);
         }
 
-        synchronized int concurrentCalls() {
-            return concurrentCalls;
+        int totalCalls() {
+            return access(() -> totalCalls);
         }
 
-        synchronized int totalCalls() {
-            return totalCalls;
+        void reset() {
+            access(() -> {
+                currentCalls = 0;
+                concurrentCalls = 0;
+                totalCalls = 0;
+                return null;
+            });
         }
 
-        synchronized void reset() {
-            currentCalls = 0;
-            concurrentCalls = 0;
-            totalCalls = 0;
+        private <T> T access(Supplier<T> operation) {
+            try {
+                accessGuard.acquire();
+                T result = operation.get();
+                accessGuard.release();
```

Comment:
This will not happen if `operation.get()` fails.

Reply:
Fixed